test/integration/test.js

```diff
@@ -111,50 +111,45 @@ const unitsNamesSimpleData = new SimpleData([
     }
 ])
 
-simpleData.mergeItems(unitsNamesSimpleData, "unit")
+simpleData.mergeItems({dataToBeMerged: unitsNamesSimpleData, commonKey: "unit"})
 
 simpleData
-    .setDefaultOptions({ nbItemInTable: "all" })
-    .addQuantiles("bonus", "salaryQuintile", 5)
-    .addBins("bonus", "salaryBins", 5)
-    .addOutliers("bonus", "bonusOutlier", "boxplot")
-    .excludeOutliers("bonus", "boxplot")
-    .setDefaultOptions({ showDataNoOverwrite: true })
-    .correlation()
-    .correlation("salary", "bonus")
+    .addQuantiles({key: "bonus", newKey: "salaryQuintile", nbQuantiles:5})
+    .addBins({key: "bonus", newKey: "salaryBins", nbBins: 5})
+    .addOutliers({key: "bonus", newKey: "bonusOutlier"})
+    .excludeOutliers({key: "bonus"})
+    .correlation() 
+    .correlation({key1: "salary", key2: "bonus"})
     .summarize()
-    .summarize(simpleData.keys, "job")
-    .summarize("salary", ["job", "unit"])
-    .summarize("salary", "job", "mean")
-    .summarize("salary", undefined, "mean")
-    .summarize("salary", "job", ["mean", "median"])
-    .summarize("salary", "job", "weightedMean", "bonus")
-
-const tempDir = temporaryDirectory()
-
-simpleData
-    .saveData(`${tempDir}/integrationTest.csv`)
-    .saveData(`${tempDir}/integrationTest.json`)
-
-simpleData.saveChart(`${tempDir}/dot1.html`, "dot", "salary", "bonus")
-simpleData.saveChart(`${tempDir}/dot2.html`, "dot", "salary", "bonus", "job")
-
-simpleData.valuesToDate("hireDate", "%Y-%m-%d")
-
-simpleData.saveChart(`${tempDir}/line1.html`, "line", "hireDate", "salary")
-simpleData.saveChart(`${tempDir}/line2.html`, "line", "hireDate", "salary", "unit")
-simpleData.saveChart(`${tempDir}/bar1.html`, "bar", "unit", "salary")
-simpleData.saveChart(`${tempDir}/bar2.html`, "bar", "unit", "salary", "unit")
-simpleData.saveChart(`${tempDir}/box1.html`, "box", "unit", "salary")
-simpleData.saveChart(`${tempDir}/box2.html`, "box", "unit", "salary", "unit")
-
-
-import * as Plot from "@observablehq/plot"
-
-simpleData
-    .saveCustomChart(
-        `${tempDir}/customChart.html`,
-        {
+    .summarize({value: simpleData.keys, key: "job"})
```

Comment:
Maybe "key" could be "groups" or "categories" ? Also, it seems that in the lines below sometimes there is key and other times keys.

Reply:
oop will fix!